internal/core/v1beta1/distribution_service.go

```diff
@@ -0,0 +1,153 @@
+package v1beta1
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kanopy-platform/cdnvalidator/internal/core"
+	"github.com/kanopy-platform/cdnvalidator/pkg/aws/cloudfront"
+	"github.com/spf13/viper"
+	"honnef.co/go/tools/config" // TODO replace with Ricardo's
+)
+
+type DistributionService struct {
+	config     *config.Config
+	configFile string
+	cloudfront *cloudfront.Client
+	awsRegion  string
+	awsKey     string
+	awsSecret  string
+	awsTimeout time.Duration
+}
+
+func New(opts ...Option) (*DistributionService, error) {
+	var err error
+
+	d := &DistributionService{
+		config:     config.New(),
+		configFile: viper.GetString("config-file"),
+		awsRegion:  viper.GetString("aws-region"),
+		awsKey:     viper.GetString("aws-key"),
+		awsSecret:  viper.GetString("aws-secret"),
+		awsTimeout: viper.GetDuration("aws-timeout"),
```

Comment:
Is it okay to use viper to get defaults?
Then have the `With..` options available so that calling code can set if they want

Reply:
viper should only ever be used in CLI packages. All internal and external packages should have configuration passed in through constructor args or functional options.